src/components/views/auth/LoginWithQR.tsx

```diff
@@ -136,16 +142,27 @@ export default class LoginWithQR extends React.Component<IProps, IState> {
                 // user denied
                 return;
             }
-            if (!this.props.client.crypto) {
+            if (!this.props.client.getCrypto()) {
                 // no E2EE to set up
                 this.props.onFinished(true);
                 return;
             }
             this.setState({ phase: Phase.Verifying });
             await this.state.rendezvous.verifyNewDeviceOnExistingDevice();
+            // clean up our state:
+            try {
+                await this.state.rendezvous.close();
+            } finally {
+                this.setState({ rendezvous: undefined });
+            }
             this.props.onFinished(true);
-        } catch (e) {
+        } catch (e: any) {
             logger.error("Error whilst approving sign in", e);
+            if (e?.httpStatus === 429) {
```

Comment:
This isn't quite right, `throw "foo"` would cause this to explode for example

```suggestion
        } catch (e) {
            logger.error("Error whilst approving sign in", e);
            if (e instanceof HTTPError && e.httpStatus === 429) {
```